client/app/scripts/utils/topology-utils.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash';
+
+export function findTopologyById(subTree, topologyId) {
+  let foundTopology;
+
+  subTree.forEach(topology => {
+    if (_.endsWith(topology.get('url'), topologyId)) {
+      foundTopology = topology;
+    }
+    if (!foundTopology && topology.has('sub_topologies')) {
+      foundTopology = findTopologyById(topology.get('sub_topologies'), topologyId);
+    }
+    if (foundTopology) {
+      return false;
+    }
+  });
+
+  return foundTopology;
+}
+
+
+export function updateNodeDegrees(nodes, edges) {
+  return nodes.map(node => {
+    const nodeId = node.get('id');
+    const degree = edges.count(edge => {
+      return edge.get('source') === nodeId || edge.get('target') === nodeId;
+    });
+    return node.set('degree', degree);
+  });
+}
+
+/* set topology.id in place on each topology */
+export function updateTopologyIds(topologies) {
+  topologies.forEach(topology => {
+    topology.id = topology.url.split('/').pop();
+    if (topology.sub_topologies) {
+      updateTopologyIds(topology.sub_topologies);
+    }
+  });
+  return topologies;
+}
```

Comment:
We could do a `fromJS` from anything that comes out of the network and make all these fns immutable too, for consistency!